components/raftstore/src/store/worker/read.rs

```diff
@@ -207,13 +213,80 @@ where
     }
 }
 
-/// #[RaftstoreCommon]: LocalReadContext combines some LocalReader's fields for temporary usage.
-pub struct LocalReadContext<'a, E>
+// If SnapshotContext is initialized with `read_id`, it manages the `SnapCache`
+// of the LocalReader. Otherwise, it fetch snapshot from KvEngine in each
+// request.
+pub struct SnapshotContext<'a, E>
 where
     E: KvEngine,
 {
-    read_id: &'a mut ThreadReadId,
-    snap_cache: &'a mut Box<Option<Arc<E::Snapshot>>>,
+    read_id: Option<ThreadReadId>,
+    snap_cache: &'a mut SnapCache<E>,
+
+    snapshot: Option<Arc<E::Snapshot>>,
```

Comment:
Why not always update `snap_cache`? What's the point of duplicated definition?

Reply:
Good suggestion.